auth/auth-store/src/main/java/com/bazaarvoice/emodb/auth/role/TableRoleManager.java

```diff
@@ -0,0 +1,280 @@
+package com.bazaarvoice.emodb.auth.role;
+
+import com.bazaarvoice.emodb.auth.permissions.PermissionIDs;
+import com.bazaarvoice.emodb.auth.permissions.PermissionManager;
+import com.bazaarvoice.emodb.auth.permissions.PermissionUpdateRequest;
+import com.bazaarvoice.emodb.common.uuid.TimeUUIDs;
+import com.bazaarvoice.emodb.sor.api.AuditBuilder;
+import com.bazaarvoice.emodb.sor.api.Coordinate;
+import com.bazaarvoice.emodb.sor.api.DataStore;
+import com.bazaarvoice.emodb.sor.api.Intrinsic;
+import com.bazaarvoice.emodb.sor.api.ReadConsistency;
+import com.bazaarvoice.emodb.sor.api.TableOptionsBuilder;
+import com.bazaarvoice.emodb.sor.api.WriteConsistency;
+import com.bazaarvoice.emodb.sor.delta.Deltas;
+import com.bazaarvoice.emodb.sor.delta.MapDeltaBuilder;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedSet;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Spliterators;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static com.bazaarvoice.emodb.common.api.Names.isLegalRoleName;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * RoleManager implementation which persists roles using tables in a {@link DataStore}.
+ */
+public class TableRoleManager implements RoleManager {
+
```

Comment:
To make it more obvious that it's the dao layer using datastore api, can we rename this class to TableRoleManagerDAO or TableDAORoleManager ? (Same with TablePermissionManager)